bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java

```diff
@@ -601,4 +602,40 @@ public static String prepareQueryForLog(String query, int maxLength) {
         ? noNewLinesQuery.substring(0, maxLength) + /* ellipsis */ '\u2026'
         : noNewLinesQuery;
   }
+
+  static String getQueryForTimePartitionedTable(
+      String destinationTableName,
+      String temporaryTableName,
+      StandardTableDefinition destinationDefinition,
+      TimePartitioning timePartitioning) {
+    TimePartitioning.Type partitionType = timePartitioning.getType();
+    String partitionField = timePartitioning.getField();
+    String extractedPartitioned = "timestamp_trunc(`%s`.`%s`, %s)";
+    String extractedPartitionedSource =
+        String.format(extractedPartitioned, "source", partitionField, partitionType.toString());
+    String extractedPartitionedTarget =
+        String.format(extractedPartitioned, "target", partitionField, partitionType.toString());
+    FieldList allFields = destinationDefinition.getSchema().getFields();
+    String commaSeparatedFields =
+        allFields.stream()
+            .map(Field::getName)
+            .map(element -> "`" + element + "`")
```

Comment:
Single joining is enough : ```.collect(Collectors.joining("`,`", "`", "`"));```

See the [API](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#joining-java.lang.CharSequence-java.lang.CharSequence-java.lang.CharSequence-)

Reply:
`.collect(Collectors.joining("`,`", "`", "`"))` will format it in the form ``field1,field2,field3`` instead of ``field1`,`field2`,`field3``